src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/activities/ActivitiesMigrationService.kt

```diff
@@ -96,7 +98,10 @@ class ActivitiesMigrationService(
 
     val allActivityIds = activitiesMappingService.getActivityMigrationDetails(migrationId, activityCount).content
       .map { it.nomisCourseActivityId }
-    nomisApiService.endActivities(allActivityIds)
+
+    val migration = migrationHistoryService.get(migrationId)
+    val activityStartDate = objectMapper.readValue(migration.filter, ActivitiesMigrationFilter::class.java).activityStartDate?.minusDays(1) ?: LocalDate.now()
+    nomisApiService.endActivities(allActivityIds, activityStartDate)
```

Comment:
The main change is to end NOMIS activities the day before the DPS activities start - if there was a start date entered (for backwards compatibility).